### PR TITLE
Use derive_new and getset macros for amethyst_utils

### PR DESCRIPTION
**Pull request by Areredify**
## Description

As per issue ([#1838]), new/get/set are replaced with corresponding macros.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Updated the content of the book if this PR would make the book outdated.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [x] Ran `cargo clippy --all --features "empty"`
- [x] Ran `cargo test --all --features "empty"`

### CODE REVIEW

**Comment by Areredify** [comment]
Is this something I did or?

File: amethyst_utils/src/auto_fov.rs [hunk 1]
```diff
@@ -19,35 +19,38 @@ use serde::{Deserialize, Serialize};
...
 
+    /// Sets `fovx_growth_rate` to the given value
+    #[set = "pub"]
     /// The factor determining how sensitive the FOV change should be
+    #[new(default)]
     fovx_growth_rate: f32,
 
     /// If the FOV grow rate specified in the field `fovx_growth_rate` should be applied as-is
+    #[new(default)]
     fixed_growth_rate: bool,
```
Comment (jojolepro) [inline amethyst_utils/src/auto_fov.rs]: I think you forgot to add the get/set macros for those?

File: amethyst_utils/src/auto_fov.rs [hunk 2]
```diff
@@ -19,35 +19,38 @@ use serde::{Deserialize, Serialize};
...
 
     /// The aspect ratio when the camera's horizontal FOV is identical to `base_fovx`
+    #[new(default)]
     base_aspect_ratio: (usize, usize),
 
     /// The minimum value the horizontal FOV can have
+    #[new(default)]
     min_fovx: f32,
```
Comment (jojolepro) [inline amethyst_utils/src/auto_fov.rs]: If you look at the Default implementation, you'll see that there's default non-zero values for those. :)

File: amethyst_utils/src/auto_fov.rs (hunk 1 above)
Comment (Areredify) [reply amethyst_utils/src/auto_fov.rs]: They have non-trivial setters and getters that return by value, so I can't add macros for those

File: amethyst_utils/src/auto_fov.rs (hunk 2 above)
Comment (Areredify) [reply amethyst_utils/src/auto_fov.rs]: Yeah, my bad

**Pull request closed without merging**